ui/pages/remote-mode/hooks/useRemoteMode.ts

```diff
@@ -0,0 +1,175 @@
+import { InternalAccount } from '@metamask/keyring-internal-api';
+import { TransactionType } from '@metamask/transaction-controller';
+import { Hex, hexToNumber } from '@metamask/utils';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  createDelegation,
+  getDeleGatorEnvironment,
+} from '../../../../shared/lib/delegation';
+import {
+  encodeDisableDelegation,
+  getDelegationHashOffchain,
+} from '../../../../shared/lib/delegation/delegation';
+import { getSelectedNetworkClientId } from '../../../../shared/modules/selectors/networks';
+import { getSelectedNetwork } from '../../../selectors';
+import { getRemoteModeConfig } from '../../../selectors/remote-mode';
+import { addTransaction } from '../../../store/actions';
+import {
+  awaitDeleteDelegationEntry,
+  listDelegationEntries,
+  signDelegation,
+  storeDelegationEntry,
+} from '../../../store/controller-actions/delegation-controller';
+import { useEIP7702Account } from '../../confirmations/hooks/useEIP7702Account';
+import { useEIP7702Networks } from '../../confirmations/hooks/useEIP7702Networks';
+import { REMOTE_MODES } from '../remote.types';
+import { useConfirmationNavigation } from '../../confirmations/hooks/useConfirmationNavigation';
+
+export const useRemoteMode = ({ account }: { account: Hex }) => {
+  const { network7702List } = useEIP7702Networks(account);
+  const globalNetworkClientId = useSelector(getSelectedNetworkClientId);
+  const selectedNetwork = useSelector(getSelectedNetwork);
+  const { chainId } = selectedNetwork.configuration;
+  const [transactionId, setTransactionId] = useState<string | undefined>();
+  const { confirmations, navigateToId } = useConfirmationNavigation();
+
+  const isRedirectPending = useMemo(() => {
+    return confirmations.some((conf) => conf.id === transactionId);
+  }, [confirmations, transactionId]);
+
+  useEffect(() => {
+    if (isRedirectPending) {
+      navigateToId(transactionId);
+    }
+  }, [isRedirectPending, navigateToId, transactionId]);
+
+  const { upgradeAccount: upgradeAccountEIP7702 } = useEIP7702Account({
+    chainId,
+  });
+
+  const remoteModeConfig = useSelector((state) =>
+    getRemoteModeConfig(state, account, chainId),
+  );
+
+  const upgradeContractAddress = useMemo(() => {
+    return getDeleGatorEnvironment(hexToNumber(chainId))
+      .EIP7702StatelessDeleGatorImpl;
+  }, [chainId]);
+
+  const delegationManagerAddress = useMemo(() => {
+    return getDeleGatorEnvironment(hexToNumber(chainId)).DelegationManager;
+  }, [chainId]);
+
+  const networkConfig = useMemo(
+    () => network7702List.find((network) => network.chainIdHex === chainId),
+    [network7702List, chainId],
+  );
+
+  const upgradeAccount = useCallback(async () => {
+    // TODO: remove this and use isSupported when it's ready
+    if (networkConfig?.isSupported) {
+      return;
+    }
```

Comment:
Not sure I understand this. We're not upgrading the account when 7702 `isSupported`?

Reply:
Oh, I see... makes sense. That naming seems confusing to me.